crates/nu-protocol/src/config.rs

```diff
@@ -65,10 +65,40 @@ impl Value {
                     config.use_ls_colors = value.as_bool()?;
                 }
                 "color_config" => {
-                    let (cols, vals) = value.as_record()?;
+                    let (cols, inner_vals) = value.as_record()?;
                     let mut hm = HashMap::new();
-                    for (k, v) in cols.iter().zip(vals) {
-                        hm.insert(k.to_string(), v.as_string()?);
+                    for (k, v) in cols.iter().zip(inner_vals) {
+                        match &v {
+                            Value::Record {
+                                cols: inner_cols,
+                                vals: inner_vals,
+                                span: _,
+                            } => {
+                                // make a string from our config.color_config section that
+                                // looks like this: { fg: "#rrggbb", bg: "#rrggbb", attr: "abc", }
+                                // the real key here was to have quotes around the values but not
+                                // require them around the keys.
+
+                                // maybe there's a better way to generate this but i'm not sure
+                                // what it is.
+                                let key = k.to_string();
+                                let mut val: String = inner_cols
+                                    .iter()
+                                    .zip(inner_vals)
+                                    .map(|(x, y)| {
+                                        let clony = y.clone();
+                                        format!("{}: \"{}\", ", x, clony.into_string(", ", &config))
+                                    })
+                                    .collect();
+                                // now insert the braces at the front and the back to fake the json string
+                                val.insert(0, '{');
+                                val.push('}');
+                                hm.insert(key, val);
```

Comment:
is there a better way to do all this fake json manipulation with serde somehow? a more idiomatic way to deserialize this particular value to regular json?

Reply:
Do you need to create a JSON string here? Could you create NuStyle directly from the Record?

Reply:
I think it needs to look like the json in the comments above this section in order to parse it later since this section is building a hashmap of string, value.

Reply:
i think we'll just run with this and see what happens